### DISCUSSION TITLE
RefTriangle3

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#278] [Q&A] [answered]
Hello @Superomeg4,
in grad_shape_function there is `if (x >= 0) or (y >= 0) or (1 - x - y >= 0):` to set the gradient.
I'm not into the math. methods of FEM, ref. elements, ... but shouldn't it be 'and' instead of 'or'?
BR S

**Comment by RaphaelPile** [comment, marked as answer]
Yeah you right. But it would be better to remove this line anyway, because it is in conflict with the tolerance introduced in "is_inside".

**Comment by RaphaelPile** [comment]
@SebGue I am going to push some improvement in the RefTriangle3 for you. You get get them in AGSF branch of EOMYS-Public.
